features/features.go

```diff
@@ -662,4 +662,13 @@ var (
 								enhancementPR(legacyFeatureGateWithoutEnhancement).
 								enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
 								mustRegister()
+
+	FeatureGateKMSv2 = newFeatureGate("KMSv2").
+				reportProblemsToJiraComponent("kube-apiserver").
+				contactPerson("swghosh").
+				productScope(kubernetes).
+				enhancementPR("https://github.com/kubernetes/enhancements/issues/3299,https://github.com/openshift/enhancements/pull/1682").
+				enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
+				enableForClusterProfile(Hypershift, configv1.Default, configv1.TechPreviewNoUpgrade).
```

Comment:
HyperShift already uses upstream kube-apiserver KMSv2 (https://github.com/openshift/hypershift/pull/4502) so we need to enable this gate by default.